### ISSUE TITLE
var_pop|stddev_pop|stddev_samp with `DISTINCT` should raise an syntax error

### ISSUE THREAD

**Issue by hidehalo** [pingcap/parser#1116]
## Bug Report

More details PTAL https://github.com/pingcap/tidb/issues/19499

**Comment by hidehalo** [comment]
@kennytm Shall we label it `type/bug`?

**Comment by kennytm** [comment]
I don't see any point *disabling* support of DISTINCT. We provide the correct result with DISTINCT, compatible with PostgreSQL.

```sql
create table a (a int);
insert into a values (1), (2), (2), (3), (3), (3);
select var_pop(all a), var_pop(distinct a), var_pop(a) from a;
-- both PostgreSQL and TiDB return 0.5555…, 0.6666…, 0.5555…
```

The additional DISTINCT keyword, although prohibited by ISO 9075*, is well supported in PostgreSQL (and CockroachDB), [BigQuery](https://cloud.google.com/bigquery/docs/reference/standard-sql/statistical_aggregate_functions), [Snowflake](https://docs.snowflake.com/en/sql-reference/functions/var_pop.html) and [Microsoft Transact-SQL](https://docs.microsoft.com/en-us/sql/t-sql/functions/varp-transact-sql?view=sql-server-ver15) and [IBM DB2](https://www.ibm.com/support/knowledgecenter/en/SSEPEK_11.0.0/sqlref/src/tpc/db2z_bif_varpop.html). MySQL (and MariaDB), Oracle, SingleStore (formerly MemSQL) followed ISO 9075 and disallowed DISTINCT. SQLite3 does not have the standard deviation/variance aggregate functions.

> ISO/IEC 9075-2:201? §4.16.4: Neither DISTINCT nor ALL are allowed to be specified for VAR_POP, VAR_SAMP, STDDEV_POP, or STDDEV_SAMP; redundant duplicates are not removed when computing these functions.

**Comment by hidehalo** [comment]
> I don't see any point _disabling_ support of DISTINCT. We provide the correct result with DISTINCT, compatible with PostgreSQL.
> 
> ```sql
> create table a (a int);
> insert into a values (1), (2), (2), (3), (3), (3);
> select var_pop(all a), var_pop(distinct a), var_pop(a) from a;
> -- both PostgreSQL and TiDB return 0.5555…, 0.6666…, 0.5555…
> ```
> 
> The additional DISTINCT keyword, although prohibited by ISO 9075*, is well supported in PostgreSQL (and CockroachDB), [BigQuery](https://cloud.google.com/bigquery/docs/reference/standard-sql/statistical_aggregate_functions), [Snowflake](https://docs.snowflake.com/en/sql-reference/functions/var_pop.html) and [Microsoft Transact-SQL](https://docs.microsoft.com/en-us/sql/t-sql/functions/varp-transact-sql?view=sql-server-ver15) and [IBM DB2](https://www.ibm.com/support/knowledgecenter/en/SSEPEK_11.0.0/sqlref/src/tpc/db2z_bif_varpop.html). Meanwhile, MySQL (and MariaDB), ~Oracle~, SingleStore (formerly MemSQL) followed ISO 9075 and disallowed DISTINCT. SQLite3 does not have the standard deviation/variance aggregate functions.
> 
> > ISO/IEC 9075-2:201? §4.16.4: Neither DISTINCT nor ALL are allowed to be specified for VAR_POP, VAR_SAMP, STDDEV_POP, or STDDEV_SAMP; redundant duplicates are not removed when computing these functions.
> 
> Edit: Oracle's non-standard [VARIANCE](https://docs.oracle.com/en/database/oracle/oracle-database/20/sqlrf/VARIANCE.html#GUID-EC33717A-2509-402D-B3BB-7EECB2E4ED8B) and [STDDEV](https://docs.oracle.com/en/database/oracle/oracle-database/20/sqlrf/STDDEV.html) functions do allow DISTINCT, but the standard ones with [_POP](https://docs.oracle.com/en/database/oracle/oracle-database/20/sqlrf/STDDEV_POP.html) and _SAMP suffixes don't 🤷.

@kennytm Yes, I have notice that TiDB provide the correct result with `DISTINCT`. 
I am not familiar with those DB&standards, this issue come from https://github.com/pingcap/tidb/issues/19499, I think maybe it is not necessary to TiDB. 
This change is about function support, I am not suitable for making decisions, maybe we should close those relative issues?
